### ISSUE TITLE
Support Autolinking, rnpm is deprecated

### ISSUE THREAD

**Issue by danilobuerger** [bugsnag/bugsnag-react-native#377]
### Description

"rnpm" is deprecated and support for it will be removed in next major version of the CLI.
See https://github.com/react-native-community/cli/blob/master/docs/configuration.md#migration-guide

### Issue

```shell
❯ yarn run react-native start
yarn run v1.16.0
$ react-native start
warn The following packages use deprecated "rnpm" config that will stop working from next release:
  - bugsnag-react-native: https://www.bugsnag.com/platforms/react-native-error-reporting/
Please notify their maintainers about it. You can find more details at https://github.com/react-native-community/cli/blob/master/docs/configuration.md#migration-guide.
```

### Environment

Library versions:

```shell
❯ yarn list react-native bugsnag-react-native react-native-code-push
yarn list v1.16.0
warning Filtering by arguments is deprecated. Please use the pattern option instead.
├─ bugsnag-react-native@2.21.1
└─ react-native@0.60.0
✨  Done in 0.58s.
```

**Comment by mattdyoung** [comment]
Hi @danilobuerger 

Thanks for bringing this to our attention. We already have a project underway to unify bugsnag-react-native with the bugsnag-js codebase and the rework will remove the use of `rnpm` so this will no longer be an issue when that is released in the near future.

**Comment by andersryanc** [comment]
@mattdyoung When this repo / project is unified with the JS one, will you still be able to enable native crash tracking? Because, isn't bugsnag-cocoa integrated as part of bugsnag-react-native? Will you have to install both (or all 3 if you include android too) separately in the future?

Also, do you have any sort of approximate timeline on that unification? Weeks? Months?

**Comment by bengourley** [comment]
Yes, native crash reporting will be supported as standard.

You will install `@bugsnag/react-native` which includes `bugsnag-cocoa` and `bugsnag-android`. When @mattdyoung says "unify", specifically he means that that the JS part of our React Native notifier will be built with the same components as our other JS notifiers.

We're working on this project right now but it's a pretty large one – it's definitely months (not weeks) away.

**Comment by danilobuerger** [comment]
@bengourley But if its months away, wouldn't it make sense to resolve the rnpm issue here? Lots of people will be hitting it with the upgrade to 0.60 and the next release 0.61 will probably be here before those months that are still needed.

**Comment by mattdyoung** [comment]
Re-opening and we'll look at resolving the deprecation issue shorter term prior to delivery of the longer term project.

**Comment by andersryanc** [comment]
I think it's a pretty straightforward and simple change. I actually started a fork for this already, but I have not yet done exhaustive testing:

https://github.com/andersryanc/bugsnag-react-native/tree/andersryanc/377-rnpm-deprecated

For reference, here is their migration guide / documentation:

https://github.com/react-native-community/cli/blob/master/docs/configuration.md#migration-guide

I did do a bit of testing around my changes, but I had to manually make those same changes inside the node_modules as I was a little unsure (based on your contributing file) how to install the package from the cloned repo... I was trying things out in the "Plain" example project in repo, but was running into some oddities which I wasn't sure if they were being caused by my change or if they might also be happening on the master branch or not... I did do a little bit of further testing on the master branch yesterday as well and still some things seemed a little off with that "Plain" project and the state it was in. So, I was thinking of running a test on a fresh RN project, but I haven't gotten there yet.
